src/Microsoft.DotNet.Build.CloudTestTasks/CreateAzureContainer.cs

```diff
@@ -1,112 +1,185 @@
         public override bool Execute()
         {
-            Log.LogMessage(MessageImportance.High, "Creating container named '{0}' in storage account {1}.", ContainerName, AccountName);
-
-            CloudStorageAccount storageAccount = CloudStorageAccount.Parse(string.Format("DefaultEndpointsProtocol=https;AccountName={0};AccountKey={1}", AccountName, AccountKey));
-            CloudBlobClient blobClient = storageAccount.CreateCloudBlobClient();
-            CloudBlobContainer newContainer = blobClient.GetContainerReference(ContainerName);
+            return this.ExecuteAsync().GetAwaiter().GetResult();
+        }
 
-            if (FailIfExists && newContainer.Exists())
+        public async Task<bool> ExecuteAsync()
+        {
+            this.Log.LogMessage(
+                MessageImportance.High, 
+                "Creating container named '{0}' in storage account {1}.", 
+                this.ContainerName, 
+                this.AccountName);
+            var dt = DateTime.UtcNow;
+            var url = string.Format(
+                "https://{0}.blob.core.windows.net/{1}?restype=container", 
+                this.AccountName, 
+                this.ContainerName);
+            this.StorageUri = string.Format(
+                "https://{0}.blob.core.windows.net/{1}/", 
+                this.AccountName, 
+                this.ContainerName);
+            var request = (HttpWebRequest)WebRequest.Create(url);
+            request.Method = "PUT";
+            request.Headers["x-ms-date"] = dt.ToString("R", CultureInfo.InvariantCulture);
+            request.Headers["x-ms-version"] = AzureHelper.StorageApiVersion;
+            request.Headers["Authorization"] = AzureHelper.AuthorizationHeader(
+                this.AccountName, 
+                this.AccountKey, 
+                "PUT", 
+                dt, 
+                request);
+
+            if (this.ReadOnlyTokenDaysValid > 0)
             {
-                Log.LogError("The container '{0}' already exists.", ContainerName);
-                return false;
+                this.ReadOnlyToken = AzureHelper.CreateContainerSasToken(
+                    this.AccountName, 
+                    this.ContainerName, 
+                    this.AccountKey, 
+                    AzureHelper.SasAccessType.read, 
+                    this.ReadOnlyTokenDaysValid);
             }
-            else
+
+            if (this.WriteOnlyTokenDaysValid > 0)
             {
-                newContainer.CreateIfNotExists();
+                this.WriteOnlyToken = AzureHelper.CreateContainerSasToken(
+                    this.AccountName, 
+                    this.ContainerName, 
+                    this.AccountKey, 
+                    AzureHelper.SasAccessType.write, 
+                    this.WriteOnlyTokenDaysValid);
             }
 
-            StorageUri = newContainer.Uri.ToString();
+            this.Log.LogMessage(MessageImportance.High, "Sending request to create Container");
```

Comment:
Probably include the container name in this log, since it's "High", it's nice to have it tell some extra information